src/context/ReactQueryCustomContext.tsx

```diff
@@ -0,0 +1,16 @@
+'use client'
+
+import { useState } from 'react'
+import { QueryClientProvider } from 'react-query'
+
+import { queryClient } from 'lib/queryClient'
+
+export const ReactQueryCustomContextProvider = ({
+  children,
+}: {
+  children: React.ReactNode
+}) => {
+  const [client] = useState(queryClient)
```

Comment:
suggestion for configuring queryClient globally
`  const [newClient] = useState(() => new QueryClient({
    queries: {
     retry: 2,
   },
 queryCache: new QueryCache({
   onError: (error, query) => {
     if (query.state.data !== undefined) {
       toast.error(error.message)
     }
   },
 }),
})) `

- 🎉 only show error toasts if we already have data in the cache which indicates a failed background update.
- retry 3 times before marking the query as an errory.
- etc ...